Sources/DistributedCluster/Plugins/ClusterSingleton/ClusterSingletonBoss.swift

```diff
@@ -155,50 +161,87 @@ internal distributed actor ClusterSingletonBoss<Act: ClusterSingleton>: ClusterS
-            if __isLocalActor(instance) {
-                self.log.debug("Passivating singleton \(instance.id)...")
-            }
-            await instance.whenLocal { __secretlyKnownToBeLocal in
-                // TODO: should have some timeout
-                await __secretlyKnownToBeLocal.passivateSingleton()
+            await instance.whenLocal { [log] in
+                log.debug("Passivating singleton \(instance.id)...")
+                // TODO: potentially do some timeout on this?
+                await $0.passivateSingleton()
             }
 
             // TODO: (optimization) tell `to` node that this node is handing off (https://github.com/apple/swift-distributed-actors/issues/329)
             // Finally, release the singleton -- it should not have been refered to strongly by anyone else,
             // causing the instance to be released. TODO: we could assert that we have released it soon after here (it's ID must be resigned).
             self.actorSystem.releaseWellKnownActorID(instance.id)
-            self.updateSingleton(nil)
+            await self.updateSingleton(nil)
         }
     }
 
-    private func updateSingleton(node: Cluster.Node?) throws {
+    /// Update on which ``Cluster/Node`` this boss considers the singleton to be hosted.
+    func updateSingleton(node: Cluster.Node?) async throws {
         switch node {
         case .some(let node) where node == self.actorSystem.cluster.node:
             // This must have been a result of an activate() and the singleton must be stored locally
             precondition(self.targetSingleton?.id.node == self.actorSystem.cluster.node)
             return
+
         case .some(let otherNode):
-            var targetSingletonID = ActorID(remote: otherNode, type: Act.self, incarnation: .wellKnown)
-            targetSingletonID.metadata.wellKnown = self.settings.name // FIXME: rather, use the BOSS as the target
-            targetSingletonID.path = self.id.path
+            var targetSingletonBossID = ActorID(remote: otherNode, type: Self.self, incarnation: .wellKnown)
+            // targetSingletonID.metadata.wellKnown = self.settings.name // FIXME: rather, use the BOSS as the target
+            targetSingletonBossID.metadata.wellKnown = Self.makeSingletonBossWellKnownName(self.settings)
+            targetSingletonBossID.path = self.id.path
+            let targetSingletonBoss = try Self.resolve(id: targetSingletonBossID, using: self.actorSystem)
+
+            let targetSingleton: Act = try await Backoff.exponential(
+                initialInterval: settings.locateActiveSingletonBackoff.initialInterval,
+                multiplier: settings.locateActiveSingletonBackoff.multiplier,
+                capInterval: settings.locateActiveSingletonBackoff.capInterval,
+                randomFactor: settings.locateActiveSingletonBackoff.randomFactor,
+                maxAttempts: settings.locateActiveSingletonBackoff.maxAttempts
+            ).attempt {
```

Comment:
`Backoff.exponential().attemtpt {}` is very nice, we're finally beginning to reap benefits of async/await inside the cluster :)